### PR TITLE
[WIP] EZP-27326: Use Nameable interface instead of deprecated getName function

### PR DESCRIPTION
**Pull request by jacek-foremski**
> JIRA : [EZP-27326](https://jira.ez.no/browse/EZP-27326)
#
Currently, Tweet FieldType uses getName() method for Content Items naming. This method is currently marked as deprecated and usage of Nameable interface is suggested instead.
Only a few core Field Types actually use Nameable interface right now, but I took them as examples.
I'm not 100% happy about testing (for example throwing RuntimeException in getName() method), but that's the way it is done in core.

**TODO:**
- [ ] Rebase to the new branches

### CODE REVIEW

File: eZ/Publish/FieldType/Tweet/NameableField.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+
+namespace EzSystems\TweetFieldTypeBundle\eZ\Publish\FieldType\Tweet;
+
+use eZ\Publish\API\Repository\Values\ContentType\FieldDefinition;
+use eZ\Publish\SPI\FieldType\Nameable;
+use eZ\Publish\SPI\FieldType\Value;
+
+class NameableField implements Nameable
```
Comment (bdunogier) [inline eZ/Publish/FieldType/Tweet/NameableField.php]: I think that I would merge this class with the Type (and tag it as both fieldtype and nameable).
In my opinion, it is easier to understand for users.

Comment (andrerom) [reply eZ/Publish/FieldType/Tweet/NameableField.php]: yes, lets keep it simple.



Note: that the deprecation on `getName` might be something to consider dropping, Nameable is optional in my mind, it's for the more advance use cases.

**Comment by jacek-foremski** [comment]
Updated per @bdunogier suggestion. Looks much simpler now!

**Review by andrerom — APPROVED** [review]

**Review by bdunogier — APPROVED** [review]

**Comment by jacek-foremski** [comment]
Ready to be merged.

**Comment by mere-teresa** [comment]
Not ready to be merged: the content of the Tutorial itself is not ready to explain the new code. (Valid for this PR, and not only).

**Pull request closed without merging**